frost-ed448/src/lib.rs

```diff
@@ -0,0 +1,405 @@
+#![allow(non_snake_case)]
+#![deny(missing_docs)]
+#![doc = include_str!("../README.md")]
+
+use ed448_goldilocks::{
+    curve::{edwards::CompressedEdwardsY, ExtendedPoint},
+    Scalar,
+};
+use rand_core::{CryptoRng, RngCore};
+use sha3::{
+    digest::{ExtendableOutput, Update, XofReader},
+    Shake256,
+};
+
+use frost_core::{frost, Ciphersuite, Field, Group};
+
+#[cfg(test)]
+mod tests;
+
+pub use frost_core::Error;
+
+#[derive(Clone, Copy)]
+/// An implementation of the FROST(Ed448, SHAKE256) ciphersuite scalar field.
+pub struct Ed448ScalarField;
+
+impl Field for Ed448ScalarField {
+    type Scalar = Scalar;
+
+    type Serialization = [u8; 57];
+
+    fn zero() -> Self::Scalar {
+        Scalar::zero()
+    }
+
+    fn one() -> Self::Scalar {
+        Scalar::one()
+    }
+
+    fn invert(scalar: &Self::Scalar) -> Result<Self::Scalar, Error> {
+        if *scalar == <Self as Field>::zero() {
+            Err(Error::InvalidZeroScalar)
+        } else {
+            Ok(scalar.invert())
+        }
+    }
+
+    fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self::Scalar {
+        Scalar::random(rng)
+    }
+
+    fn serialize(scalar: &Self::Scalar) -> Self::Serialization {
+        let bytes = scalar.to_bytes();
+        std::array::from_fn(|i| if i < 56 { bytes[i] } else { 0 })
```

Comment:
Is this a workaround until the upstream fixes are merged?

Reply:
Ed448 is tricky because points are encoded as 57 bytes (56 full bytes for the field element, forcing an extra byte for the sign). But scalars can be encoded as 56 bytes. I'm working on a PR upstream to make `Scalar::to_bytes()` return 57 bytes but it does make it weird because Ed448-Goldilocks also supports X448 and in that scenario there is no reason to encode scalars as 57 bytes and all test vectors have 56 bytes :smiling_face_with_tear: 

Anyway I'll check with the maintainer what seems to be the best solution, feedback is welcome

Reply:
Weird, not my favorite, but oh well!